src/common/helpers/db-tls-validate.sh

```
@@ -0,0 +1,337 @@
+
+    if [ -z "$ips" ]; then
+        # If hostname doesn't resolve, try direct connection to hostname
+        local direct_cmd="openssl s_client -connect $host:$port"
+        [ -f "$pem_path" ] && direct_cmd="$direct_cmd -CAfile $pem_path"
+        case "$db_type" in
+            postgres) direct_cmd="$direct_cmd -starttls postgres" ;;
+            mysql|mariadb) direct_cmd="$direct_cmd -starttls mysql" ;;
+        esac
+
+        log "$LOG_CONTEXT" "ℹ️" "No DNS resolution for ${host}, testing direct connection"
+        if echo "Q" | timeout 2s $direct_cmd >/dev/null 2>&1; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)